src/unix/aix.c

```diff
@@ -1152,3 +1155,56 @@ void uv__platform_invalidate_fd(uv_loop_t* loop, int fd) {
   if(loop->backend_fd >= 0)
     pollset_ctl(loop->backend_fd, &pc, 1);
 }
+
+int uv__enable_largesocket(uv_stream_t *stream) {
+  struct optreq1 oreq;
+  int ws;
+  int rs;
+  int rmem;
+  int wmem;
+  int rr;
+  int rw;
+
+  int fd = socket(1, 1, 0);
+  if (fd < 0)
+    return UV_ENOSYS;
+
+  oreq.getnext = 1;
+  rs = 0;
+  ws = 0;
+  rmem = 0;
+  wmem = 0;
+  while (ioctl(fd, SIOCGNETOPT1, (caddr_t)&oreq) != -1) {
+    if (strncmp(oreq.name, "tcp_sendspace", strlen("tcp_sendspace")) == 0) {
+      wmem = atoi(oreq.data);
+      ws = 1;
+      if (rs == 1)
+        break;
+    }
+    if (strncmp(oreq.name, "tcp_recvspace", strlen("tcp_recvspace")) == 0) {
+      rmem = atoi(oreq.data);
+      rs = 1;
+      if (ws == 1)
+        break;
+    }
+  }
+
+  uv__close(fd);
+
+  if (rs == 0 || ws == 0)
+    return UV_ENOSYS;
+
+  if (rmem <= (64 * 1024) || wmem <= (64 * 1024))
+    return UV_ENOSYS;
+
+  rr = uv_recv_buffer_size((uv_handle_t*) stream, &rmem);
+  if (rr != 0)
+    return UV_ENOSYS;
+
+  rw = uv_send_buffer_size((uv_handle_t*) stream, &wmem);
```

Comment:
done, retained only one (renamed to `r` , following conventions on return values)